### PR TITLE
En_Al

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Al/z_en_al.c [hunk 1]
```diff
@@ -18,100 +19,892 @@ void EnAl_Draw(Actor* thisx, GlobalContext* globalCtx);
...
+
+    Math_Vec3f_Copy(&sp40, &this->unk_368->world.pos);
+    Math_Vec3f_Copy(&sp34, &this->actor.world.pos);
+
+    Math_ApproachS(&this->unk_4DE, Math_Vec3f_Yaw(&sp34, &sp40) - this->actor.shape.rot.y, 4, 0x2AA8);
+    this->unk_4DE = CLAMP(this->unk_4DE, -0x1FFE, 0x1FFE);
+
+    Math_Vec3f_Copy(&sp34, &this->actor.focus.pos);
+
+    if (this->unk_368->id == 0) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Al/z_en_al.c:496]:
```suggestion
    if (this->unk_368->id == ACTOR_PLAYER) {
```

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_En_Al/z_en_al.h [hunk 2]
```diff
@@ -6,12 +6,43 @@
...
+    /* 0x0190 */ MtxF unk_190[6];
+    /* 0x0310 */ ColliderCylinder unk_310;
+    /* 0x035C */ u8 unk_35C;
+    /* 0x0360 */ s32* unk_360;
+    /* 0x0364 */ s32 unk_364;
+    /* 0x0368 */ Actor* unk_368;
+    /* 0x036C */ Vec3f unk_36C;
+    /* 0x0378 */ Vec3s unk_378;
+    /* 0x037E */ Vec3s jointTable[27];
+    /* 0x0420 */ Vec3s morphTabke[27];
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Al/z_en_al.h:26]:
```suggestion
    /* 0x0420 */ Vec3s morphTable[27];
```
typo

File: src/overlays/actors/ovl_En_Al/z_en_al.c [hunk 3]
```diff
@@ -18,100 +19,895 @@ void EnAl_Draw(Actor* thisx, GlobalContext* globalCtx);
...
+            break;
+
+        case 16:
+            Matrix_MultiplyVector3fByState(&D_80BE0070, &this->actor.focus.pos);
+            Math_Vec3s_Copy(&this->actor.focus.rot, &this->actor.world.rot);
+            break;
+    }
+}
+
+void EnAl_TransformDraw(GlobalContext* globalCtx, s32 limbIndex, Actor* thisx) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Al/z_en_al.c:860]:
```suggestion
void EnAl_TransformLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Actor* thisx) {
```
Would prefer including Limb to match the other limb draws (override and post).

Comment (hensldm) [reply src/overlays/actors/ovl_En_Al/z_en_al.c]: I don't remember saying `TransformDraw` by itself, but if I did, that would have been my mistake. My bad.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**